### DISCUSSION TITLE
Websocket reverse-proxy nginx

### DISCUSSION THREAD

**Discussion by sulthanalihsan** [serverpod/serverpod#744] [Q&A]
I just deployed my serverpod project to vps, everything perfectly running well, from running endpoints, authentication, till sending verification email, but I have problem when using realtime functionality (websocket) especially when setting up a server with reverse-proxy nginx, I can't connect to websocket with error code 400, here is the error I found

![image](https://user-images.githubusercontent.com/30169687/224547472-4ae9003a-bbb7-47fd-a312-c42180aa0bfc.png)

![image](https://user-images.githubusercontent.com/30169687/224547481-46d65ac7-3630-4986-b522-34a9b1ec4e56.png)

what should I do, please, thanks

**Comment by vlidholt** [comment]
I'm moving this to the discussions as we currently only have official support for AWS (and very soon GCP).

Hopefully, someone else can help!

**Reply by braj065** [reply]
Is it possible to have documentation in future on how to setup and run spod on VPS?

**Comment by hortigado** [comment]
Friend, could you help with a tutorial on how you managed to do the implementation on a vps? I have not been able to achieve it

**Comment by lukecastronuevo** [comment]
I'm a bit late but here's my NGINX configuration that made it work over HTTPS: 
```
server_name api.examplepod.com;

location / {
       proxy_pass http://localhost:8080;
}

# Proxy WebSocket traffic
location /websocket {
        proxy_pass http://your.server.ip.here:8080;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
}
```
Make sure to configure your SSL Certificates

**Reply by gnotus** [reply]
you saved my websockets

**Reply by HaroldRomero151199** [reply]
para las nuevas actualizaciones seria añadir /api1/websocket
`location /v1/websocket {`
